lib/blob/cluster/cluster.go

```diff
@@ -228,23 +232,24 @@ func (c *cluster) fetchNewObjects() error {
 			missingObjects = append(missingObjects, hash)
 		}
 	}
+	var errors []error
 	for _, hash := range missingObjects {
 		c.Infof("Found missing object %v.", hash)
 		err = c.fetchObject(hash)
 		if err != nil {
-			c.Warningf("Failed to fetch object(%v) %v.", hash, trace.DebugReport(err))
-			return trace.Wrap(err)
+			c.WithError(err).Warnf("Failed to fetch object(%v).", hash)
```

Comment:
Nit: Use field for the hash? `c.WithError(err).WithField("hash", hash).Warn()`

Also - do you know if there's some other information about the missing object available here (in addition to hash) that we can add to the log as well to make it more useful in determining which package this refers to? Countless times we've seen "failed to fetch object(qweasdzxc)" in the logs without a slightest idea of what package this is. It's probably not available here as blob is low level but I'm just wondering.

Reply:
No, but we can add `metadata` field so there's a way to trace a blob to a specific package (anywhere in the cluster).